### PR TITLE
CountAsLoss + fix CheckStartSolo3v3Arena

### PR DESCRIPTION
**Pull request by laasker**
this changes need https://github.com/azerothcore/azerothcore-wotlk/commit/f6de601ddd5218290268268ee695c56d88847225 (merged already)

## Changes Proposed:
- config 'Solo.3v3.StopGameIncomplete' now works, before it would always end the arena despite havin 6 players
- now applies Deserter if a player doesn't accept queue, or logout while queue pops
- now count as a loss if a player leaves arena or doesn't accept queue
- statistics option now shows solo MMR

## Issues Addressed:

- Closes

## Tests Performed:

- Tested ingame
- 

## How to Test the Changes:

1.
2.
3.

### CODE REVIEW

File: src/solo3v3_sc.h [hunk 1]
```diff
@@ -66,7 +66,6 @@ class Solo3v3BG : public AllBattlegroundScript
     uint32 oldTeamRatingHorde;
 
     void OnQueueUpdate(BattlegroundQueue* queue, uint32 /*diff*/, BattlegroundTypeId bgTypeId, BattlegroundBracketId bracket_id, uint8 arenaType, bool isRated, uint32 /*arenaRatedTeamId*/) override;
-    void OnBattlegroundUpdate(Battleground* bg, uint32 /*diff*/) override;
```
Comment (Helias) [inline src/solo3v3_sc.h:69]: why did you delete this?

Comment (laasker) [reply src/solo3v3_sc.h:69]: We have the hook OnArenaStart now, so we can use it to call CheckStartSolo3v3.

But we could use OnBGUpdate and check if there's a player that didn't accept the queue and then end the solo earlier, while not in progress, instead of waiting the arena to start

**Comment by Helias** [comment]
pipeline error
```
/home/runner/work/mod-arena-3v3-solo-queue/mod-arena-3v3-solo-queue/modules/mod-arena-3v3-solo-queue/src/solo3v3_sc.cpp:670:14: fatal error: use of undeclared identifier 'ARENA_DESERTION_TYPE_LEAVE_BG'; did you mean 'BG_DESERTION_TYPE_LEAVE_BG'?
  670 |         case ARENA_DESERTION_TYPE_LEAVE_BG:
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |              BG_DESERTION_TYPE_LEAVE_BG
/home/runner/work/mod-arena-3v3-solo-queue/mod-arena-3v3-solo-queue/src/server/game/Battlegrounds/Battleground.h:50:5: note: 'BG_DESERTION_TYPE_LEAVE_BG' declared here
   50 |     BG_DESERTION_TYPE_LEAVE_BG          = 0, // player leaves the BG
      |     ^
1 error generated.
```

**Comment by Helias** [comment]
minor issue
```
/home/runner/work/mod-arena-3v3-solo-queue/mod-arena-3v3-solo-queue/modules/mod-arena-3v3-solo-queue/src/solo3v3.cpp:144:24: fatal error: comparison of integers of different signs: 'int' and 'uint32' (aka 'unsigned int') [-Wsign-compare]
  144 |     if (PlayersInArena < AmountPlayersSolo3v3)
      |         ~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [modules/CMakeFiles/modules.dir/build.make:104: modules/CMakeFiles/modules.dir/mod-arena-3v3-solo-queue/src/solo3v3.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
```

**Comment by laasker** [comment]
> minor issue
> 
> ```
> /home/runner/work/mod-arena-3v3-solo-queue/mod-arena-3v3-solo-queue/modules/mod-arena-3v3-solo-queue/src/solo3v3.cpp:144:24: fatal error: comparison of integers of different signs: 'int' and 'uint32' (aka 'unsigned int') [-Wsign-compare]
>   144 |     if (PlayersInArena < AmountPlayersSolo3v3)
>       |         ~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> gmake[2]: *** [modules/CMakeFiles/modules.dir/build.make:104: modules/CMakeFiles/modules.dir/mod-arena-3v3-solo-queue/src/solo3v3.cpp.o] Error 1
> gmake[2]: *** Waiting for unfinished jobs....
> ```

My bad, it's fixed now
